src/pages/workspaces/workspace/analysis/cost-utils.js

```diff
@@ -0,0 +1,233 @@
+  const defaultNodepoolIpAddressCost = ephemeralExternalIpAddressCost({ numStandardVms: 1, numPreemptibleVms: 0 })
+
+  const staticCost = defaultNodepoolComputeCost + defaultNodepoolIpAddressCost
+  const dynamicCost = app.kubernetesRuntimeConfig.numNodes *
+    getHourlyCostForMachineType(app.kubernetesRuntimeConfig.machineType, defaultComputeRegion, false) +
+    ephemeralExternalIpAddressCost({ numStandardVms: app.kubernetesRuntimeConfig.numNodes, numPreemptibleVms: 0 })
+
+  switch (appStatus) {
+    case 'STOPPED':
+      return staticCost
+    case 'DELETING':
+    case 'ERROR':
+      return 0.0
+    default:
+      return staticCost + dynamicCost
+  }
+}
+
+/*
+ * - Disk cost is incurred regardless of app status.
+ * - Disk cost is total for data (NFS) disk, metadata (postgres) disk, and boot disks (1 boot disk per nodepool)
+ * - Size of a data disk is user-customizable. The other disks have fixed sizes.
+ */
+export const getGalaxyDiskCost = ({ size: dataDiskType, diskType }) => {
+  const metadataDiskSize = 10 // GB
+  const defaultNodepoolBootDiskSize = 100 // GB
+  const appNodepoolBootDiskSize = 100 // GB
+
+  return getPersistentDiskCostHourly({
+    status: 'Running',
+    size: dataDiskType + metadataDiskSize + defaultNodepoolBootDiskSize + appNodepoolBootDiskSize,
+    diskType
+  }, defaultComputeRegion)
+}
+
+export const getGalaxyCostTextChildren = (app, appDataDisks) => {
+  const dataDisk = getCurrentAttachedDataDisk(app, appDataDisks)
+  return app ?
+    [getComputeStatusForDisplay(app.status), dataDisk ? ` ${Utils.formatUSD(getGalaxyCost(app, dataDisk))}/hr` : ''] : ['']
+}
+
+// end GOOGLE COST METHODS
+
+// AZURE COST METHODS begin
+
+export const getAzureComputeCostEstimate = ({ region, machineType }) => {
+  const regionPriceObj = getAzurePricesForRegion(region)
+  const cost = regionPriceObj[machineType]
+  return cost
+}
+
+export const getAzureDiskCostEstimate = ({ region, diskSize }) => {
+  const regionPriceObj = getAzurePricesForRegion(region)
+  const diskType = getDiskType(diskSize)
+  const cost = regionPriceObj[diskType]
+  return cost
+}
+
+// end AZURE COST METHODS
+
+// COMMON METHODS begin
+
+export const getPersistentDiskCostMonthly = ({ cloudContext = {}, diskType, size, status, zone }, computeRegion) => {
+  let price
+  if (isAzureContext(cloudContext)) {
+    price = getAzureDiskCostEstimate({ diskSize: size, region: zone })
+  } else {
+    price = size * getPersistentDiskPriceForRegionMonthly(computeRegion, diskType)
+  }
+  return _.includes(status, ['Deleting', 'Failed']) ? 0.0 : price
+}
+export const getPersistentDiskCostHourly = ({ size, status, diskType, cloudContext = {} }, computeRegion) => {
+  const price = isAzureContext(cloudContext) ? getAzureDiskCostEstimate({ diskSize: size, region: computeRegion }) / numberOfHoursPerMonth : getPersistentDiskPriceForRegionHourly(computeRegion, diskType)
+  return _.includes(status, ['Deleting', 'Failed']) ? 0.0 : size * price
```

Comment:
I think this calculation is incorrect. getAzureDiskCostEstimate is the complete monthly estimate. However, we're additionally multiplying size to that final figure. We may want to move the size multiplier next to getPersistentDiskPriceForRegionHourly.

Reply:
![Screenshot 2023-02-07 at 10 13 37 AM](https://user-images.githubusercontent.com/11773357/217284417-e93e00ca-0c7a-4bf6-a89d-4266ed98866e.png)
![Screenshot 2023-02-07 at 10 13 42 AM](https://user-images.githubusercontent.com/11773357/217284419-7d03abe3-7bec-4107-bb34-30076e1e566a.png)

Reply:
Done! added to the GCP case in getPersistentDiskCostHourly - other methods rely on getPdPriceForRegionHourly so I didn't want to change its output.